src/components/text.js

```diff
@@ -0,0 +1,372 @@
+
+  'aileronsemibold': FONT_BASE_URL + 'Aileron-Semibold.fnt',
+  'dejavu': FONT_BASE_URL + 'DejaVu-sdf.fnt',
+  'exo2bold': FONT_BASE_URL + 'Exo2Bold.fnt',
+  'exo2semibold': FONT_BASE_URL + 'Exo2SemiBold.fnt',
+  'kelsonsans': FONT_BASE_URL + 'KelsonSans.fnt',
+  'monoid': FONT_BASE_URL + 'Monoid.fnt',
+  'mozillavr': FONT_BASE_URL + 'mozillavr.fnt',
+  'sourcecodepro': FONT_BASE_URL + 'SourceCodePro.fnt'
+};
+
+var loadedFontPromises = {};
+var loadedTexturePromises = {};
+
+coreShader.registerShader('modified-sdf', {
+  schema: {
+    alphaTest: {type: 'number', is: 'uniform', default: 0.5},
+    color: {type: 'color', is: 'uniform', default: 'white'},
+    map: {type: 'map', is: 'uniform'},
+    opacity: {type: 'number', is: 'uniform', default: 1.0}
+  },
+  vertexShader: [
+    'varying vec2 vUV;',
+    'void main(void) {',
+    '  gl_Position = projectionMatrix * modelViewMatrix * vec4(position, 1.0);',
+    '  vUV = uv;',
+    '}'
+  ].join('\n'),
+  fragmentShader: [
+    '#define ALL_SMOOTH 0.5',
+    '#define ALL_ROUGH 0.4',
+    '#define DISCARD_ALPHA 0.1',
+    'uniform sampler2D map;',
+    'uniform vec3 color;',
+    'uniform float opacity;',
+    'uniform float alphaTest;',
+    'varying vec2 vUV;',
+    'float aastep(float value) {',
+    '  float afwidth = (1.0 / 32.0) * (1.4142135623730951 / (2.0 * gl_FragCoord.w));',
+    '  return smoothstep(0.5 - afwidth, 0.5 + afwidth, value);',
+    '}',
+    'void main() {',
+    '  vec4 texColor = texture2D(map, vUV);',
+    '  float value = texColor.a;',
+    '  float alpha = aastep(value);',
+    '  float ratio = (gl_FragCoord.w >= ALL_SMOOTH) ? 1.0 : (gl_FragCoord.w < ALL_ROUGH) ? 0.0 : (gl_FragCoord.w - ALL_ROUGH) / (ALL_SMOOTH - ALL_ROUGH);',
+    '  if (alpha < alphaTest) { if (ratio >= 1.0) { discard; return; } alpha = 0.0; }',
+    '  alpha = alpha * ratio + (1.0 - ratio) * value;',
+    '  if (ratio < 1.0 && alpha <= DISCARD_ALPHA) { discard; return; }',
+    '  gl_FragColor = vec4(color, opacity * alpha);',
+    '}'
+  ].join('\n')
+});
+
+module.exports.Component = registerComponent('text', {
+  schema: {
+    align: {type: 'string', default: 'left', oneOf: alignments},
+    alphaTest: {default: 0.5},
+    // center default to match primitives like plane; if 'align', null or undefined, same as align
+    anchor: {default: 'center', oneOf: anchors},
+    baseline: {default: 'center', oneOf: baselines},
+    color: {type: 'color', default: '#000'},
+    customShader: {default: 'modified-sdf', oneOf: shaderNames},
+    font: {type: 'string', default: 'default'},
+    // default to fnt but with .fnt replaced by .png
+    fontImage: {type: 'string'},
+    // no default, will be populated at layout
+    height: {type: 'number'},
+    letterSpacing: {type: 'number', default: 0},
+    // default to font's lineHeight value
+    lineHeight: {type: 'number'},
+    opacity: {type: 'number', default: '1.0'},
+    shader: {default: 'custom', oneOf: ['custom', 'sdf', 'basic', 'msdf']},
```

Comment:
I do think there are use cases for custom shaders (e.g. this hybrid modification, or outlines, or interesting text effects), in which case they should be from the registered shader list, which is why I originally chose an explicitly verifiable "custom" setting for `shader` (essentially a promise you know what you are doing) and then `customShader` uses the existing registered shader name list